bears/general/InvalidLinkBear.py

```diff
@@ -85,12 +89,18 @@ def find_links_in_file(file, timeout, link_ignore_regex, link_ignore_list):
                         fnmatch(link, link_ignore_list)):
                     if link.startswith(('hg+', 'bzr+', 'git+', 'svn+')):
                         link = InvalidLinkBear.parse_pip_vcs_url(link)
-                    code = InvalidLinkBear.get_status_code(link, timeout)
+                    host = urlparse(link).netloc
+                    code = InvalidLinkBear.get_status_code(link,
+                                                           network_timeout[
+                                                               host])
                     yield line_number + 1, link, code
 
+    return_default_timeout = lambda: DEFAULT_TIMEOUT  # pragma: no cover
+
     @deprecate_settings(link_ignore_regex='ignore_regex')
     def run(self, filename, file,
-            timeout: int=DEFAULT_TIMEOUT,
+            network_timeout: typed_dict(str, int, DEFAULT_TIMEOUT)=
+            defaultdict(return_default_timeout),
```

Comment:
Explanation for not using `lambda` directly here: 
- Because the coverage wasn't counted correctly, since the `lambda` was **never** called according to `coverage`, so I moved it out and marked it a `no cover` outside, because marking it here would lead to marking the whole function as `no cover` while it is completely covered